src/asm/fstack.c

```diff
@@ -115,6 +124,29 @@ static int32_t popcontext(void)
 			sym_UseCurrentMacroArgs();
 			sym_SetMacroArgID(nMacroCount++);
 			sym_UseNewMacroArgs();
+
+			/* Increment REPT count in file path */
+			pREPTIterationWritePtr =
+				strrchr(tzCurrentFileName, '~') + 1;
+			nREPTIterationNo =
+				strtoul(pREPTIterationWritePtr, NULL, 10);
+			nNbCharsLeft = sizeof(tzCurrentFileName)
+				- (pREPTIterationWritePtr - tzCurrentFileName);
+			nNbCharsWritten = snprintf(pREPTIterationWritePtr,
+						   nNbCharsLeft, "%lu",
+						   nREPTIterationNo + 1);
+			if (nNbCharsWritten > nNbCharsLeft) {
+				/*
+				 * The string is not not terminated anymore,
```

Comment:
`snprintf()` always appends a null terminator, so the string will be terminated. I think this should say: "The string is truncated,".

Reply:
Reading `snprintf`'s manpage, I see no mention of that, only that "at most `size` bytes will be written". Worst case, this can be changed to "The string hasn't been properly written" to be more catch-all.

Reply:
I've changed the first line to "The string is probably corrupted somehow," which should be correct and catch-all enough.

Reply:
That's fine. According to the C11 standard, `snprintf()` has to write the null terminator as long as `n` is not zero. From 7.21.6.5/2:
> If **n** is zero, nothing is written, and **s** may  be  a  null  pointer.   Otherwise,  output  characters  beyond  the **n-1**st  are discarded rather than being written to the array, and a null character is written at the end of  the  characters  actually  written  into  the  array.

You're right that Linux man pages sound a bit confusing.
> The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str.

I am pretty sure they are trying to say the same thing as the standard, but they didn't express it very well.

Reply:
"Including **a** terminating null byte" would probably be clearer.